### PR TITLE
Add student search bar on Admin panel

### PR DESCRIPTION
**Pull request by webecke**
## Overview

Add a search bar to the student tab on the admin panel, and also update the About page.

## Testing

- [ ] Added/updated unit tests
- [ ] Tested edge cases
- [X] Manual testing (if needed)

image
image

I requested a billion people to review this, but I just need one person. Whoever wants to do it just go for it.

### CODE REVIEW

File: src/main/resources/frontend/src/components/AboutPage.vue [hunk 1]
```diff
@@ -95,21 +97,21 @@ onUnmounted(() => {
...
-          contributions="Created several admin tools, cleaned up the Student UI and added Font Awesome"
+          contributions="Created several admin tools, cleaned up the Student UI, built a config system, and added Font Awesome"
           fa-icon="fa-solid fa-tree"
         />
         <AboutPagePerson
           name="James Finlinson"
           url="https://github.com/frozenfrank"
-          tenure="Jan 2024-Dec 2025"
+          tenure="Jan 2024-Apr 2025"
           contributions="Developed class-requirement enforcement systems, like late-days and git-commits"
```
Comment (frozenfrank) [inline src/main/resources/frontend/src/components/AboutPage.vue:107]: I would like to mention my GitHub Project contributions here

**Review by frozenfrank — COMMENTED** [review]
Looks pretty good to me.

Have we manually tested what happens on the page when searching, leaving the page, and returning? We may want to restore the same search term, or reset it, but it's something we want to decide. All the links currently open up modals, so a single user can navigate between students prettily easily, right?

**Comment by webecke** [comment]
> Have we manually tested what happens on the page when searching, leaving the page, and returning? 

Yes, the search term stays when looking at other tabs, but disappears when you reload the page. This is intended behavior.

>  All the links currently open up modals, so a single user can navigate between students prettily easily, right?

Clicking on a cell of the table opens up modals, clicking directly on the link opens the link in a new tab

File: src/main/resources/frontend/src/components/AboutPage.vue (hunk 1 above)
Comment (webecke) [reply src/main/resources/frontend/src/components/AboutPage.vue:107]: How would you phrase it?

"Developed class-requirement enforcement systems like late-days and git-commits, and built dev-ops tools"

Up to you

**Review by 19mdavenport — APPROVED** [review]
Looks good to me whenever you have all the about pages sorted
